### PR TITLE
MODORG-3 - Restrict search/view of Organization records based upon acquisitions unit

### PR DESCRIPTION
**Pull request by siarhei-charniak**
[MODORG-3](https://issues.folio.org/browse/MODORG-3) - Restrict search/view of Organization records based upon acquisitions unit

## Purpose
Restrict search/view of Organization based upon the acquisitions unit memberships of the user and the acquisitions unit being assigned to the record.

## Approach
* Acquisitions units are used to determine if a user can search for/view the organization record
* Unit tests are updated

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [x] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -163,6 +168,31 @@
         </executions>
       </plugin>
 
+      <plugin>
+        <groupId>org.jsonschema2pojo</groupId>
+        <artifactId>jsonschema2pojo-maven-plugin</artifactId>
+        <version>0.5.1</version>
```
Comment (grabsefx) [inline pom.xml:174]: the latest version is 1.0.2

File: src/main/java/org/folio/service/protection/ProtectionServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  private AcquisitionsUnitsService acquisitionsUnitsService;
+  private final List<AcquisitionsUnit> fetchedUnits = new ArrayList<>();
+
+  @Autowired
+  public void setAcquisitionsUnitsService(AcquisitionsUnitsService acquisitionsUnitsService) {
+    this.acquisitionsUnitsService = acquisitionsUnitsService;
+  }
+
+  @Override
+  public CompletableFuture<Void> isOperationRestricted(List<String> unitIds, Set<ProtectedOperationType> operations, String lang, Context context, Map<String, String> headers) {
```
Comment (grabsefx) [inline src/main/java/org/folio/service/protection/ProtectionServiceImpl.java:37]: method name and return type confuses a bit. Having "isOperationRestricted" means that method will return Boolean value, but returns Void. Please use more meaningful naming or change returning type

Comment (siarhei-charniak) [reply src/main/java/org/folio/service/protection/ProtectionServiceImpl.java]: renamed

File: pom.xml (hunk 1 above)
Comment (siarhei-charniak) [reply pom.xml]: fixed

**Pull request merged**